src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -548,7 +548,7 @@ private bool CheckHash(bool compatMode)
                 }
                 case SubjectIdentifierType.SubjectKeyIdentifier:
                 {
-                    filtered = extraStore.Find(X509FindType.FindBySubjectKeyIdentifier, signerIdentifier.Value, false);
+                    filtered = extraStore.Find(X509FindType.FindBySubjectKeyIdentifier, signerIdentifier.Value!, false);
```

Comment:
What guarantees Value is non-null?

Reply:
The type isn't NoSignature (the only time Value is null)

Reply:
At some time in the future, if we ever figured out a way to compute this, it'd be interesting to understand just how many unnecessary null checks users end up adding to their code based on the (correct) annotations we've added. :(

Reply:
Yeah, "this is null only when some other local state says it will be" and "this isn't null when some other local state says it won't be" are definitely big holes in the logic.

Effectively this type is a discriminated union; and assuming you read the typeID field you now know that the value can't be null.

Similarly with the cert.Get[Alg]PublicKey() methods... if you asked the "what kind of thing is this" you already know the nullarity.  We made them return null for the wrong algorithm for simpler flow control (and annoying cases like redundant identifiers).